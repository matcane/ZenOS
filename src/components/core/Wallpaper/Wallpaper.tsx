import { PropsWithChildren } from "react";
import { ImageBackground } from "react-native";

import { coreStyles } from "@/styles/core";

import image from "@assets/images/bg.webp";

const { wallpaper } = coreStyles;

export default function Wallpaper({ children }: PropsWithChildren) {
  return (
    <ImageBackground testID="image-background" source={image} resizeMode="cover" style={wallpaper}>
      {children}
    </ImageBackground>
  );
}
