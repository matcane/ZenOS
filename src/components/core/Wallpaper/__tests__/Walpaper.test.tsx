import { render } from "@testing-library/react-native";
import { Text } from "react-native";

import image from "@assets/images/bg.webp";

import Wallpaper from "../Wallpaper";

jest.mock("@assets/images/bg.webp", () => ({
  uri: "mocked-image-uri",
}));

jest.mock("expo-router", () => ({
  useSegments: jest.fn(() => []),
}));

describe("Wallpaper component", () => {
  it("renders correctly", () => {
    const { toJSON } = render(<Wallpaper />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders children inside ImageBackground", () => {
    const { getByText } = render(
      <Wallpaper>
        <Text>Test Child</Text>
      </Wallpaper>,
    );

    expect(getByText("Test Child")).toBeTruthy();
  });

  it("sets the correct background image", () => {
    const { getByTestId } = render(
      <Wallpaper>
        <Text>Test Child</Text>
      </Wallpaper>,
    );

    const imageBackground = getByTestId("image-background");
    expect(imageBackground.props.source).toEqual(image);
  });
});
