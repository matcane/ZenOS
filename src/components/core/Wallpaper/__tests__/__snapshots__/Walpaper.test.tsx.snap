// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallpaper component renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "uri": "mocked-image-uri",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="image-background"
  />
</View>
`;
